dora/services/utils.py

```diff
@@ -215,3 +216,26 @@ def filter_services_by_region(services, region_code):
             & Q(diffusion_zone_details=region.code)
         )
     )
+
+
+def compute_map_bounds(geom: MultiPolygon | Point, distance: int):
+    # Crée les objets de référence spatiale pour la transformation
+    original_srs = SpatialReference(geom.srid)
+    target_srs = SpatialReference(3857)  # Projection de Mercator (en mètres)
```

Comment:
Mercator est pas top, elle déforme énormément les distances. La "bonne" façon de le faire, c’est de trouver une bonne projection locale (c’est ce que fait PostGIS par exemple : https://postgis.net/docs/ST_Buffer.html)
Si on avait pas les DROM ce serait simple, 2154 partout… Mais c’est pas le cas.
Après c’est pas grave de pas être super précis dans ce cas, mais je ne me rends pas compte de la taille de l’erreur qu’on fait là…

Suggestion:
- tester en France metropolitaine et dans les differents DROM, si le résultat est satisfaisant en l’état 
- sinon chercher une meilleure projection générique ; y a pas de solution miracle, mais on peut trouver un truc suffisant (plate carrée peut-être ?)
- sinon calculer une projection à la volée (voir https://gis.stackexchange.com/questions/121489/1km-circles-around-lat-long-points-in-many-places-in-world/121539#121539 par ex)
- sinon, laisser tomber tout ça, ne renvoyer que le point ou la geom, et calculer le buffer en front avec https://turfjs.org/docs/#buffer suivi de https://turfjs.org/docs/#bbox mais il faut vérifier que ça n’ajoute pas une dépendance trop lourde sur le front
- ou trouver une lib en python qui fasse ça correctement :sweat_smile: 

Mais surtout, à la reflexion, le mieux je me dis c’est de ne pas calculer de buffer du tout...
Pour les recherches precises, tu renvoies juste un point, pour les recherches de ville tu renvoies juste geom.extent. 
Coté front, si tu as un point tu fais un flyTo ou un jumpTo sur le point, avec un zoom genre 17, si tu as une bbox, tu fais un fitBounds sur la bbox, puis un getZoom pour récupérer le zoom actuel, et tu dezoome de deux niveaux avec setZoom

Reply:
On pourrait pas choisir la projection dynamiquement en fonction de la taille de la box ? Style prendre une proj Lambert93 si on se cantonne à une zone/box France, et une autre (WGS84?) si on a du DOM/TOM ?

Reply:
Très intéressant. Merci pour toutes ces idées !

Effectivement, vouloir zoomer de façon précise sur la zone de recherche et relève de l'over-engineering, d'autant plus que la limite de 50 km est valable aussi bien pour les recherches sur adresse et sur une ville.

Donc autant jouer sur le niveau de zoom.

J'ai retiré le code de calcul de la zone pour ne retourner que les limites de la ville (de9bc1c79d84329c708e73dd322fc9ef332daa8c).

Les coordonnées en cas d'une recherche sur adresse sont déjà disponibles côté front-end.

Implémentation sur le front : https://github.com/gip-inclusion/dora-front/pull/405/commits/411344e7e3de4709adb9c2869feffd8b8f1b7c00

Reply:
> On pourrait pas choisir la projection dynamiquement en fonction de la taille de la box

oui le lien stackexchange que je proposais fait ça, avec une projection dynamique en fonction de la lat/long.
Sinon on peut avoir une liste de correspondance pour chaque DROM de la projection officielle, mais c’est quand même pas mal compliqué pour pas grand chose je pense.

Reply:
> J'ai retiré le code de calcul de la zone pour ne retourner que les limites de la ville

ça me parait l’approche la plus simple vu l’usage, oui !